staging/src/k8s.io/apiserver/pkg/admission/plugins.go

```diff
@@ -140,6 +140,7 @@ func (ps *Plugins) NewFromPlugins(pluginNames []string, configProvider ConfigPro
 			return nil, err
 		}
 		if plugin != nil {
+			plugin = NewFieldManagerAdmissionWrapper(plugin, pluginName)
```

Comment:
I think the best way to plumb this part through it to reuse the decorator pattern that we already have for admission controller metrics, ex: https://github.com/kubernetes/kubernetes/blob/016e9d5c06089774c6286fd825302cbae661a446/staging/src/k8s.io/apiserver/pkg/server/options/admission.go#L79
(defined [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/admission/metrics/metrics.go#L51-L54)) this would wrap each named admission controller and receive the name of that controller, without making a special case for the FieldManager admission wrapper